presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1379,8 +1380,14 @@ private static HiveStorageFormat extractHiveStorageFormat(Table table)
 
     private static Function<HiveColumnHandle, ColumnMetadata> columnMetadataGetter(Table table, TypeManager typeManager)
     {
+        Set<String> partitionKeyNames = table.getPartitionKeys().stream().map(FieldSchema::getName).collect(Collectors.toSet());
+        List<FieldSchema> tableColumns = table.getSd().getCols();
+        if (tableColumns.stream().anyMatch(field -> partitionKeyNames.contains(field.getName()))) {
+            throw new PrestoException(HIVE_INVALID_METADATA, "Partition column cannot also be a table column");
```

Comment:
The issue is that we build a map immediately afterwards, which fails when we have two entries with the same key. @haozhun said that partition columns should not appear in the table, which is why I chose this error message.